### ISSUE TITLE
Fedora Silverblue support ?

### ISSUE THREAD

**Issue by GoldenHashtag** [morrownr/8821cu#27]
`Silverblue` has read-only root filesystem, so modifying existing system directly is impossible:
```
Running install-driver.sh version 20210524
Starting installation.
Copying source files to: /usr/src/rtl8821cu-5.8.1.7
cp: cannot create directory '/usr/src/rtl8821cu-5.8.1.7': Read-only file system
Copying 8821cu.conf to: /etc/modprobe.d
Error! Could not find module source directory.
Directory: /usr/src/rtl8821cu-5.8.1.7 does not exist.
An error occurred. dkms add error = 2
Please report this error.
```

To properly install out of tree modules on `Silverblue` you had to use `rpm-ostree install`. But 8821cu doesn't provide a RPM file, nor any build instruction for RPM.
Also, `Silverblue` prefers `akmod` for dynamically rebuilding modules on kernel changing.
Related issue: https://github.com/coreos/rpm-ostree/issues/1091

**Comment by GoldenHashtag** [comment]
never mind. i've purchased a new adapter with in tree driver

**Comment by morrownr** [comment]
I'm a little behind these days. I was pondering a solution for this but you will probably be happier with an adapter using an in-kernel driver anyway.

**Comment by GoldenHashtag** [comment]
yeah, I picked up COMFAST CF-WU785AC based on your guide: [The Short List - Superstar USB WiFi Adapters for Linux](https://github.com/morrownr/USB-WiFi/blob/main/The_Short_List.md)
and it's working great out of the box, plug in and play.

**Comment by morrownr** [comment]
If you are interested, write up a review and post it in Issues at the main USB-WiFi

https://github.com/morrownr/USB-WiFi

Whatever you think Linux users would like to know.

**Comment by GoldenHashtag** [comment]
I would love to. But it just works™, without any issues or quirks.
I did have issue with Gnome Wi-Fi settings handling multiple interfaces poorly. The solution is to use `nm-connection-editor` instead. Maybe you add this to USB-WiFi page?

**Comment by morrownr** [comment]
> I would love to. But it just works™, without any issues or quirks.

People like to read about what is good. Write something up when you have time. My experience with the 785AC so far is just like you said, it "just works."

> I did have issue with Gnome Wi-Fi settings handling multiple interfaces poorly. The solution is to use `nm-connection-editor` instead. Maybe you add this to USB-WiFi page?

I'm not a fan of gnome so I had not noticed this. What problem were you seeing?

**Comment by GoldenHashtag** [comment]
> People like to read about what is good. Write something up when you have time. My experience with the 785AC so far is just like you said, it "just works."

👌

> What problem were you seeing?

- When you toggle Wi-Fi switch, all Wi-Fi interfaces are turned on/off, even though the toggles are displayed per Wi-Fi interface, causing confusions.
- You can't disconnect a Wi-Fi, clicking on connected access point only makes it reconnect, you can make it disconnect by forgetting it but then passwords and configurations are gone. Combining with above issue, you can't make it so one adapter connects to a Wi-Fi while another disconnects without forgetting connection.
- Wi-Fi Hotspot always uses the first Wi-Fi interface, sorted by the time it was detected. You can't specifically choose a interface to use hotspot.
- The only good thing is that Gnome supports WPS push, wheres KDE doesn't.

**Comment by GoldenHashtag** [comment]
To disconnect a Wi-Fi one requires an extension: [Disconnect Wifi](https://extensions.gnome.org/extension/904/disconnect-wifi/)
This is Gnome℠.

**Comment by morrownr** [comment]
Oh, I am familiar with these issues. How about we all go start filing bugs about the poor interface and configurability. The bugs could be filed with Ubuntu, gnome, Fedora and a few other places. That stuff needs to be fixed.

I use Linux Mint with Cinnamon as my main driver. It has nm-connection-editor built in. I wear it out as you would image given what I do here.

I have tried to love KDE and I do like it more than gnome but it just kills me how buggy it is. I try it once per year just to see how it is going but bugs, bugs and more bugs... and it is heavier than I like also.

I have tried to like gnome but there are two things that drive me away 1) It is a slow pig. And 2) the desktop metaphor just does not work for me.

Funny how we managed to get so far off topic.

**Comment by GoldenHashtag** [comment]
For Gnome there had been issues talking about them:
- https://gitlab.gnome.org/GNOME/gnome-control-center/-/issues/112
- https://gitlab.gnome.org/GNOME/gnome-control-center/-/issues/791
So I don't think that simply submitting issues can help resolving them, one has to actually submit PR.

Also have you found a way to package akmod RPM file? The closest I found was [this](https://rpmfusion.org/Packaging/KernelModules/Kmods2) where it talked about using a kmod src RPM. I believe you have to use RPM-based distro to create one (damn the lack of standardization), and since you're using Mint, you may not want to do so.
I have zero experience developing kernel module, but if you kinda instruct somehow then I could try to sort this out myself.
Not that I'm rushing you, given the fact I've been using new adapter anyway. But having an additional adapter for hotspot would be cool.

**Comment by morrownr** [comment]
> For Gnome there had been issues talking about them:
> 
> * https://gitlab.gnome.org/GNOME/gnome-control-center/-/issues/112
> * https://gitlab.gnome.org/GNOME/gnome-control-center/-/issues/791
>   So I don't think that simply submitting issues can help resolving them, one has to actually submit PR.

Not everyone has to be able to code. Adding information to an existing issue or starting a new issue that is well written makes sure the issue is on the agenda. Squeaky wheel gets the grease.

> Also have you found a way to package akmod RPM file? 

No. I am not familiar with that package manager.

> and since you're using Mint, you may not want to do so.

I really don't have time to additional things right now.

> I have zero experience developing kernel module, but if you kinda instruct somehow then I could try to sort this out myself.

This could work and I am more than happy to help. Warning: This is not easy. Over a year ago, when COVID-19 was going strong and there was no vaccine, I needed something to keep me busy and away from people so I decided to try to help in the area of usb wifi. I first looked at the code for the rtl88x2bu. I got an adapter with a 8812bu chipset and dusted off my old C books and got started. Five drivers and nine supported chipsets later, you see what you see now at this site. Linux users with existing hardware needed help. However, the most important repo that I have created is probably the USB-WiFi. It became obvious to me after a while that most Linux users did not realize that there were usb wifi adapters available that did not require the pain and suffering that these Realtek adapters put us through.

Tell me what is on your mind and I will start pointing you in the right direct to find the information you need.

**Comment by GoldenHashtag** [comment]
It's been a week since then, time flies.

so, TLDR:
**After repeatedly attempting switching to Linux for years, I've now gave up Linux and new adapter was returned, switched back to Windows with previous 8811CU adapter, mainly because of the Linux community atmosphere.**

<details>
  <summary>Rant warning, you may as well just pass this to save you time</summary>

I originally typed plenty of plenty of words (much more than following), then decided to remove them.

Linux Desktop has many issues, yet most people in the community decides to ignore them and show hostility towards those who criticism it. You have freedom to modify, to redistribute, to sell free software, you don't have freedom to criticism it.
The example: [Major Linux Problems  on the Desktop](https://itvision.altervista.org/why.linux.is.not.ready.for.the.desktop.current.html) and [Linux (In)security](https://madaidans-insecurities.github.io/linux.html) got bashed by community, instead of trying to solve them.
Also try to google for:

- `Windows site:reddit.com/r/linuxmasterrace`
- `Windows site:reddit.com/r/linux`

And you'll see the worst of community keep bitching on Windows and praising Linux.

------
Praising Linux or diminishing Windows won't help making Linux better, it only makes developer pass Linux as a whole because `why make Linux software when no one will buy it?`  `why fix things that ain't broke?`, and as a result, no Adobe or MS Office (And Linux community loves to tell that GIMP/LibreOffice is a valuable alternative when reality they're absolutely not)

These quotes from [Major Linux Problems on the Desktop](https://itvision.altervista.org/why.linux.is.not.ready.for.the.desktop.current.html) summarize my rant:
> It's worth noting that the most vocal participants of the Open Source community are extremely bitchy and overly idealistic people peremptorily requiring everything to be open source and free or it has no right to exist at all in Linux.With an attitude like this, it's no surprise that a lot of companies completely disregard and shun the Linux desktop.

> Even though the Open Source movement lacks manpower, different Linux distros find enough resources to fork projects (Gentoo developers are going to develop a udev alternative; a discord in ffmpeg which led to the emergence of libav; a situation around OpenOffice/LibreOffice; a new X.org/Wayland alternative - Mir) and to use their own solutions. 

As well as [this](https://itvision.altervista.org/why-linux-gnu-might-never-succeed.html#top).



> Not everyone has to be able to code. Adding information to an existing issue or starting a new issue that is well written makes sure the issue is on the agenda. 

Linux community [don't think so](https://joshuastrobl.com/2021/09/14/building-an-alternative-ecosystem/):

> if you don’t contribute, your opinion doesn’t matter

Also [this](http://disq.us/p/2i5i08l) is what I encountered as well:

> Q: I use this feature X and I have problem Y with it, is there any fix?
> A1: Use feature O, it's better (it's not & I didn't even asked for it)
> A2: Who even uses feature X? (Me and a lot of users use it, but you don't care)
> A3: I don't even have a problem with it, delay is quite good, idk what you're talking about. (tell me more sweet little lies pls)
> A4: That is not a Linux problem at all, that is how it should function (If it should function badly, then sure)
> A5: You didn't plugged the device D right, hence you have the problem Y (I did plugged it right. If it isn't plugged right, than whole device wouldn't even work. You even know this, but hey, why not making more troll posts. It's user's fault always, but not Linux's)
> A6: I will give you the solution to the problem E, which is not directly related to the problem Y, but still try this fix [I'm not responsible for the termonuclear war & any problem you may face even If I gave a bad advice. Remember, It's your fault, not Linux's] (you do the solution to the problem E, you can't boot to system anymore when that so called solution is applied, you get the F comments afterwards in the forum section + the comment from the legendary solutionist who says: Ooops, I don't care. You should read the fucking manual.)

------

After all the tinkers with Linux I've got really, really tired. I'm even tired to rant anything, I just want a working system, which Linux Desktop failed to provide to me. I've encountered system lockup on heavy IO (Hint: [solution ](https://askubuntu.com/questions/397249/system-freezes-unresponsive-unusable-when-copying-large-file-to-usb) didn't work for me), combining with the community atmosphere, this is the showstopper for me, I'm quitting.
So TLDR of TLDR: Linux community sucks. I've went back to Windows to avoid the community as much as possible. 

------

</details>

You may close this issue now since I don't need Silverblue support anymore.

------

> People like to read about what is good. Write something up when you have time.

Now I do have some words on this: `CF-WU785AC` works terribly on Windows when combining with proxy, I don't know why but when using with proxy, the connection gets dropped frequently (Wi-Fi connection still there, but application keeps complaining `forcibly closed by remote host` and browser downloads get interrupted all the times), direct connections are unaffected, there might be something wrong on my end but I'm tired of troubleshooting. Also on Windows it's not Plug and Play, instead a virtual CD with driver installer appears on first plug in, in comparison my previous Realtek 8811CU has true Plug and Play support on Windows and no connection issue with proxy. So I returned it after switching back to Windows.
